spec/lib/danger/ci_sources/support/pull_request_finder_spec.rb

```diff
@@ -0,0 +1,117 @@
+require "danger/ci_source/support/pull_request_finder"
+
+RSpec.describe Danger::PullRequestFinder do
+  def finder(pull_request_id: "", logs: nil, repo_slug: "danger/danger", remote: "false")
+    described_class.new(
+      pull_request_id,
+      repo_slug,
+      remote: remote,
+      git_logs: logs
+    )
+  end
+
+  def merge_pull_request_log
+    IO.read("spec/fixtures/ci_source/support/danger-git.log")
+  end
+
+  def squash_and_merge_log
+    IO.read("spec/fixtures/ci_source/support/swiftweekly.github.io-git.log")
+  end
+
+  def two_kinds_of_merge_log
+    IO.read("spec/fixtures/ci_source/support/two-kinds-of-merge-both-present.log")
+  end
+
+  def open_pull_requests_info
+    require "ostruct"
+    JSON.parse(
+      IO.read("spec/fixtures/ci_source/support/danger_danger_pr_518.json"),
+      object_class: OpenStruct
```

Comment:
This is a TIL for me, the hash can use like objects.

http://stackoverflow.com/a/32470168